pkg/http/handler/timeout.go

```diff
@@ -75,8 +75,9 @@ func (h *timeToFirstByteTimeoutHandler) ServeHTTP(w http.ResponseWriter, r *http
 		h.handler.ServeHTTP(tw, r.WithContext(ctx))
 	}()
 
-	timeout := time.NewTimer(h.timeout)
-	defer timeout.Stop()
+	timeout := getTimer(h.timeout)
+	var timeoutExpired bool
```

Comment:
I think this will be a bit clearer if `timeoutExpired` is renamed to `timeoutDrained` (which is already the param name in putTimer) since that's the key thing going on here, and associates the variable here with the one in the method, which is important to understand why this works

Reply:
Ack, changed.